### PR TITLE
Dockerfile: update to Fedora 40

### PR DESCRIPTION
**Pull request by aaradhak**
https://github.com/coreos/fedora-coreos-tracker/issues/1674

### CODE REVIEW

**Comment by aaradhak** [comment]
@jlebon I see that the release notes are already updated to Fedora 40 in this case - https://github.com/coreos/butane/blob/main/docs/release-notes.md

**Comment by jlebon** [comment]
> @jlebon I see that the release notes are already updated to Fedora 40 in this case - [`main`/docs/release-notes.md](https://github.com/coreos/butane/blob/main/docs/release-notes.md?rgh-link-date=2024-06-12T21%3A09%3A27Z)

Not quite. That release note entry refers to the `butane` binary. The container image itself being based on Fedora 40 is a separate change. Though it doesn't look like historically we've release noted it.

That said, this is missing another instance in the same file.

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM registry.fedoraproject.org/fedora:39 AS builder
+FROM registry.fedoraproject.org/fedora:40 AS builder
```
Comment (prestist) [inline Dockerfile:1]: Do this but also on the next "FROM" line.

**Comment by prestist** [comment]
@aaradhak  after you have updated both From's you can then add a label for "skip notes" as we dont generally doc the update of this.

File: Dockerfile (hunk 1 above)
Comment (aaradhak) [reply Dockerfile:1]: missed that one. Thanks for spotting that.

**Review by jlebon — APPROVED** [review]

**Pull request merged**